// smooth-ui/internal/createComponent.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _handleRef = _interopRequireDefault(require("./handleRef"));

var _setWithComponent = _interopRequireDefault(require("./setWithComponent"));

var defaultTheme = _interopRequireWildcard(require("../theme/defaultTheme"));

var utils = _interopRequireWildcard(require("../utils"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    ", ";\n    ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function createComponent(getConfig) {
  var _getConfig = getConfig(_objectSpread({}, utils, {
    css: _styledComponents.css,
    classNames: _classnames.default
  })),
      name = _getConfig.name,
      style = _getConfig.style,
      _getConfig$defaultPro = _getConfig.defaultProps,
      defaultProps = _getConfig$defaultPro === void 0 ? {} : _getConfig$defaultPro,
      _getConfig$propTypes = _getConfig.propTypes,
      propTypes = _getConfig$propTypes === void 0 ? {} : _getConfig$propTypes,
      _getConfig$render = _getConfig.render,
      render = _getConfig$render === void 0 ? function (_ref) {
    var Component = _ref.Component,
        props = _objectWithoutPropertiesLoose(_ref, ["Component"]);

    return _react.default.createElement(Component, props);
  } : _getConfig$render,
      _getConfig$defaultCom = _getConfig.defaultComponent,
      defaultComponent = _getConfig$defaultCom === void 0 ? 'div' : _getConfig$defaultCom,
      InnerComponentFromConfig = _getConfig.InnerComponent;

  var InnerComponent = InnerComponentFromConfig || function (_ref2) {
    var className = _ref2.className,
        _ref2$component = _ref2.component,
        Component = _ref2$component === void 0 ? defaultComponent : _ref2$component,
        theme = _ref2.theme,
        props = _objectWithoutPropertiesLoose(_ref2, ["className", "component", "theme"]);

    return render(_objectSpread({
      Component: Component,
      className: (0, _classnames.default)("sui-" + name, className)
    }, props));
  };

  InnerComponent.displayName = "sui-" + name;
  var RefComponent = (0, _handleRef.default)(InnerComponent);
  var StyledComponent = (0, _styledComponents.default)(RefComponent)(_templateObject(), utils.mixin('base'), style);
  StyledComponent.propTypes = _objectSpread({
    theme: _propTypes.default.object
  }, propTypes);
  StyledComponent.defaultProps = _objectSpread({
    theme: defaultTheme
  }, defaultProps);
  (0, _setWithComponent.default)(StyledComponent, RefComponent);
  return StyledComponent;
}

var _default = createComponent;
exports.default = _default;
