// smooth-ui/internal/InnerSwitch.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var defaultTheme = _interopRequireWildcard(require("../theme/defaultTheme"));

var _utils = require("../utils");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 100%;\n  height: 100%;\n  position: absolute;\n  z-index: ", ";\n\n  .sui-inner-switch-input {\n    margin: 0;\n    padding: 0;\n    position: absolute;\n    height: 100%;\n    width: 100%;\n    opacity: 0;\n    z-index: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var InnerSwitchContainer = _styledComponents.default.div(_templateObject(), (0, _utils.th)('zIndexInnerSwitch'), (0, _utils.th)('zIndexInnerSwitch'));

var InnerSwitch =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(InnerSwitch, _React$Component);

  function InnerSwitch(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;

    _this.handleChange = function (event) {
      if (_this.props.checked === undefined) {
        _this.updateState({
          checked: event.target.checked
        });
      }

      if (_this.context.suiGroup) {
        _this.context.suiGroup.notify(event);
      }

      if (_this.props.onChange) {
        _this.props.onChange(event);
      }
    };

    _this.handleFocus = function (event) {
      _this.updateState({
        focused: true
      });

      if (_this.props.onFocus) {
        _this.props.onFocus(event);
      }
    };

    _this.handleBlur = function (event) {
      _this.updateState({
        focused: false
      });

      if (_this.props.onBlur) {
        _this.props.onBlur(event);
      }
    };

    _this.state = {
      focused: false,
      checked: props.checked || props.defaultChecked || false
    };
    return _this;
  }

  var _proto = InnerSwitch.prototype;

  _proto.componentWillMount = function componentWillMount() {
    if (this.context.suiGroup) {
      this.context.suiGroup.register(this);
    }

    if (this.props.onStateChange) {
      this.props.onStateChange(this.state);
    }
  };

  _proto.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if (nextProps.checked !== undefined && nextProps.checked !== this.state.checked) {
      this.updateState({
        checked: nextProps.checked
      });
    }
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    if (this.context.suiGroup) {
      this.context.suiGroup.unregister(this);
    }
  };

  _proto.updateState = function updateState(state) {
    if (this.props.onStateChange) {
      this.props.onStateChange(state);
    }

    this.setState(state);
  };

  _proto.render = function render() {
    var _this$props = this.props,
        baseRef = _this$props.baseRef,
        checked = _this$props.checked,
        children = _this$props.children,
        defaultChecked = _this$props.defaultChecked,
        inputType = _this$props.inputType,
        onStateChange = _this$props.onStateChange,
        onChange = _this$props.onChange,
        onBlur = _this$props.onBlur,
        onFocus = _this$props.onFocus,
        theme = _this$props.theme,
        props = _objectWithoutPropertiesLoose(_this$props, ["baseRef", "checked", "children", "defaultChecked", "inputType", "onStateChange", "onChange", "onBlur", "onFocus", "theme"]);

    return _react.default.createElement(InnerSwitchContainer, {
      theme: theme,
      className: "sui-inner-switch"
    }, _react.default.createElement("input", _extends({
      ref: baseRef,
      className: "sui-inner-switch-input",
      type: inputType,
      onChange: this.handleChange,
      checked: this.state.checked,
      onFocus: this.handleFocus,
      onBlur: this.handleBlur
    }, props)), this.props.children({
      focused: this.state.focused,
      checked: this.state.checked,
      disabled: props.disabled
    }));
  };

  return InnerSwitch;
}(_react.default.Component);

InnerSwitch.propTypes = {
  checked: _propTypes.default.bool,
  children: _propTypes.default.func,
  defaultChecked: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  inputType: _propTypes.default.string,
  onChange: _propTypes.default.func,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func,
  onStateChange: _propTypes.default.func,
  theme: _propTypes.default.object
};
InnerSwitch.defaultProps = {
  theme: defaultTheme
};
InnerSwitch.contextTypes = {
  suiGroup: _propTypes.default.object
};
var _default = InnerSwitch;
exports.default = _default;
