// smooth-ui/theme/defaultTheme.js
"use strict";

exports.__esModule = true;
exports.yiqContrastedThreshold = exports.breakpoints = exports.transitionBase = exports.zIndexModalBackdrop = exports.zIndexModal = exports.zIndexInnerSwitch = exports.zIndexControl = exports.alertColorLevel = exports.alertBorderLevel = exports.alertBgLevel = exports.alertMarginBottom = exports.alertPaddingX = exports.alertPaddingY = exports.controlFocusBoxShadow = exports.controlFocusBorderColor = exports.inputTextColor = exports.inputPlaceholderText = exports.inputDisabledText = exports.inputDisabledBgColor = exports.inputBgColor = exports.inputBorderColor = exports.inputBorderWidth = exports.inputLineHeightLg = exports.inputPaddingXLg = exports.inputPaddingYLg = exports.inputLineHeightSm = exports.inputPaddingXSm = exports.inputPaddingYSm = exports.inputLineHeight = exports.inputPaddingX = exports.inputPaddingY = exports.btnDisabledOpacity = exports.btnBorderWidth = exports.btnLineHeightLg = exports.btnPaddingXLg = exports.btnPaddingYLg = exports.btnLineHeightSm = exports.btnPaddingXSm = exports.btnPaddingYSm = exports.btnLineHeight = exports.btnPaddingX = exports.btnPaddingY = exports.inputBtnBorderWidth = exports.inputBtnLineHeightLg = exports.inputBtnPaddingXLg = exports.inputBtnPaddingYLg = exports.inputBtnLineHeightSm = exports.inputBtnPaddingXSm = exports.inputBtnPaddingYSm = exports.inputBtnLineHeight = exports.inputBtnPaddingX = exports.inputBtnPaddingY = exports.lineHeightLg = exports.lineHeightSm = exports.lineHeightBase = exports.fontWeightBold = exports.fontWeightNormal = exports.fontWeightLight = exports.fontSizeLg = exports.fontSizeSm = exports.fontSizeBase = exports.borderWidth = exports.borderRadiusLg = exports.borderRadiusSm = exports.borderRadius = exports.yikTextLight = exports.yikTextDark = exports.secondaryLight = exports.primaryLight = exports.dark = exports.light = exports.danger = exports.warning = exports.info = exports.success = exports.secondary = exports.primary = exports.cyan = exports.teal = exports.green = exports.yellow = exports.orange = exports.brick = exports.red = exports.pink = exports.purple = exports.indigo = exports.blue = exports.black = exports.gray900 = exports.gray800 = exports.gray700 = exports.gray600 = exports.gray500 = exports.gray400 = exports.gray300 = exports.gray200 = exports.gray100 = exports.white = exports.fontFamily = void 0;
exports.colorYik = exports.colorLevel = exports.alertVariant = exports.btnVariant = exports.controlFocus = exports.base = exports.modalFooterBorderWidth = exports.modalHeaderBorderWidth = exports.modalFooterBorderColor = exports.modalHeaderBorderColor = exports.modalContentBoxShadowSmUp = exports.modalContentBoxShadowXs = exports.modalContentBorderRadius = exports.modalContentBorderColor = exports.modalContentBorderWidth = exports.modalContentBg = exports.modalDialogMarginYSmUp = exports.modalDialogMargin = exports.modalTransitionDuration = exports.modalInnerPadding = exports.modalBackdropBg = exports.displayLineHeight = exports.display4Weight = exports.display3Weight = exports.display2Weight = exports.display1Weight = exports.display4Size = exports.display3Size = exports.display2Size = exports.display1Size = exports.h6FontSize = exports.h5FontSize = exports.h4FontSize = exports.h3FontSize = exports.h2FontSize = exports.h1FontSize = exports.headingsColor = exports.headingsLineHeight = exports.headingsFontWeight = exports.headingsFontFamily = exports.headingsMarginBottom = exports.colorInterval = void 0;

var _styledComponents = require("styled-components");

var _polished = require("polished");

var _defaultBreakpoints = _interopRequireDefault(require("./defaultBreakpoints"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n    color: ", ";\n    background-color: ", ";\n    border-color: ", ";\n\n    hr {\n      border-top-color: ", ";\n    }\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  background-color: ", ";\n\n  &:focus {\n    ", ";\n  }\n\n  &:not(:disabled):hover,\n  &:not(:disabled):active {\n    background-color: ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  outline: 0;\n  box-shadow: 0 0 2px ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  box-sizing: border-box;\n  font-family: ", ";\n  font-size: ", ";\n  line-height: ", ";\n\n  * {\n    box-sizing: border-box;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["0 .5rem 1rem ", ""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["0 .25rem .5rem ", ""]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["0 0 0 0.2rem ", ""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

// Fonts
var fontFamily = '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif'; // Colors

exports.fontFamily = fontFamily;
var white = '#fff';
exports.white = white;
var gray100 = '#f8f9fa';
exports.gray100 = gray100;
var gray200 = '#e9ecef';
exports.gray200 = gray200;
var gray300 = '#dee2e6';
exports.gray300 = gray300;
var gray400 = '#ced4da';
exports.gray400 = gray400;
var gray500 = '#adb5bd';
exports.gray500 = gray500;
var gray600 = '#6c757d';
exports.gray600 = gray600;
var gray700 = '#495057';
exports.gray700 = gray700;
var gray800 = '#343a40';
exports.gray800 = gray800;
var gray900 = '#212529';
exports.gray900 = gray900;
var black = '#000';
exports.black = black;
var blue = '#007bff';
exports.blue = blue;
var indigo = '#6610f2';
exports.indigo = indigo;
var purple = '#6f42c1';
exports.purple = purple;
var pink = '#e83e8c';
exports.pink = pink;
var red = '#dc3545';
exports.red = red;
var brick = '#bd4932';
exports.brick = brick;
var orange = '#fd7e14';
exports.orange = orange;
var yellow = '#ffc107';
exports.yellow = yellow;
var green = '#28a745';
exports.green = green;
var teal = '#20c997';
exports.teal = teal;
var cyan = '#17a2b8';
exports.cyan = cyan;
var primary = brick;
exports.primary = primary;
var secondary = gray600;
exports.secondary = secondary;
var success = green;
exports.success = success;
var info = cyan;
exports.info = info;
var warning = yellow;
exports.warning = warning;
var danger = red;
exports.danger = danger;
var light = gray100;
exports.light = light;
var dark = gray800;
exports.dark = dark;
var primaryLight = (0, _utils.th)('primary', function (c) {
  return (0, _polished.lighten)(0.3, c);
});
exports.primaryLight = primaryLight;
var secondaryLight = (0, _utils.th)('secondary', function (c) {
  return (0, _polished.lighten)(0.3, c);
});
exports.secondaryLight = secondaryLight;
var yikTextDark = '#111';
exports.yikTextDark = yikTextDark;
var yikTextLight = '#fff'; // Borders

exports.yikTextLight = yikTextLight;
var borderRadius = '.25rem';
exports.borderRadius = borderRadius;
var borderRadiusSm = '.2rem';
exports.borderRadiusSm = borderRadiusSm;
var borderRadiusLg = '.3rem';
exports.borderRadiusLg = borderRadiusLg;
var borderWidth = '1px'; // Fonts

exports.borderWidth = borderWidth;
var fontSizeBase = '1rem';
exports.fontSizeBase = fontSizeBase;
var fontSizeSm = (0, _utils.th)('fontSizeBase', function (fontSize) {
  return (0, _polished.modularScale)(-1, fontSize);
});
exports.fontSizeSm = fontSizeSm;
var fontSizeLg = (0, _utils.th)('fontSizeBase', function (fontSize) {
  return (0, _polished.modularScale)(1, fontSize);
});
exports.fontSizeLg = fontSizeLg;
var fontWeightLight = 300;
exports.fontWeightLight = fontWeightLight;
var fontWeightNormal = 400;
exports.fontWeightNormal = fontWeightNormal;
var fontWeightBold = 700;
exports.fontWeightBold = fontWeightBold;
var lineHeightBase = 1.5;
exports.lineHeightBase = lineHeightBase;
var lineHeightSm = 1.5;
exports.lineHeightSm = lineHeightSm;
var lineHeightLg = 1.5; // Input + Buttons

exports.lineHeightLg = lineHeightLg;
var inputBtnPaddingY = '.375rem';
exports.inputBtnPaddingY = inputBtnPaddingY;
var inputBtnPaddingX = '.75rem';
exports.inputBtnPaddingX = inputBtnPaddingX;
var inputBtnLineHeight = (0, _utils.th)('lineHeightBase');
exports.inputBtnLineHeight = inputBtnLineHeight;
var inputBtnPaddingYSm = '.25rem';
exports.inputBtnPaddingYSm = inputBtnPaddingYSm;
var inputBtnPaddingXSm = '.5rem';
exports.inputBtnPaddingXSm = inputBtnPaddingXSm;
var inputBtnLineHeightSm = (0, _utils.th)('lineHeightSm');
exports.inputBtnLineHeightSm = inputBtnLineHeightSm;
var inputBtnPaddingYLg = '.5rem';
exports.inputBtnPaddingYLg = inputBtnPaddingYLg;
var inputBtnPaddingXLg = '1rem';
exports.inputBtnPaddingXLg = inputBtnPaddingXLg;
var inputBtnLineHeightLg = (0, _utils.th)('lineHeightLg');
exports.inputBtnLineHeightLg = inputBtnLineHeightLg;
var inputBtnBorderWidth = (0, _utils.th)('borderWidth'); // Buttons

exports.inputBtnBorderWidth = inputBtnBorderWidth;
var btnPaddingY = (0, _utils.th)('inputBtnPaddingY');
exports.btnPaddingY = btnPaddingY;
var btnPaddingX = (0, _utils.th)('inputBtnPaddingX');
exports.btnPaddingX = btnPaddingX;
var btnLineHeight = (0, _utils.th)('inputBtnLineHeight');
exports.btnLineHeight = btnLineHeight;
var btnPaddingYSm = (0, _utils.th)('inputBtnPaddingYSm');
exports.btnPaddingYSm = btnPaddingYSm;
var btnPaddingXSm = (0, _utils.th)('inputBtnPaddingXSm');
exports.btnPaddingXSm = btnPaddingXSm;
var btnLineHeightSm = (0, _utils.th)('inputBtnLineHeightSm');
exports.btnLineHeightSm = btnLineHeightSm;
var btnPaddingYLg = (0, _utils.th)('inputBtnPaddingYLg');
exports.btnPaddingYLg = btnPaddingYLg;
var btnPaddingXLg = (0, _utils.th)('inputBtnPaddingXLg');
exports.btnPaddingXLg = btnPaddingXLg;
var btnLineHeightLg = (0, _utils.th)('inputBtnLineHeightLg');
exports.btnLineHeightLg = btnLineHeightLg;
var btnBorderWidth = 0;
exports.btnBorderWidth = btnBorderWidth;
var btnDisabledOpacity = 0.8; // Inputs

exports.btnDisabledOpacity = btnDisabledOpacity;
var inputPaddingY = (0, _utils.th)('inputBtnPaddingY');
exports.inputPaddingY = inputPaddingY;
var inputPaddingX = (0, _utils.th)('inputBtnPaddingX');
exports.inputPaddingX = inputPaddingX;
var inputLineHeight = (0, _utils.th)('inputBtnLineHeight');
exports.inputLineHeight = inputLineHeight;
var inputPaddingYSm = (0, _utils.th)('inputBtnPaddingYSm');
exports.inputPaddingYSm = inputPaddingYSm;
var inputPaddingXSm = (0, _utils.th)('inputBtnPaddingXSm');
exports.inputPaddingXSm = inputPaddingXSm;
var inputLineHeightSm = (0, _utils.th)('inputBtnLineHeightSm');
exports.inputLineHeightSm = inputLineHeightSm;
var inputPaddingYLg = (0, _utils.th)('inputBtnPaddingYLg');
exports.inputPaddingYLg = inputPaddingYLg;
var inputPaddingXLg = (0, _utils.th)('inputBtnPaddingXLg');
exports.inputPaddingXLg = inputPaddingXLg;
var inputLineHeightLg = (0, _utils.th)('inputBtnLineHeightLg');
exports.inputLineHeightLg = inputLineHeightLg;
var inputBorderWidth = (0, _utils.th)('inputBtnBorderWidth');
exports.inputBorderWidth = inputBorderWidth;
var inputBorderColor = (0, _utils.th)('gray300');
exports.inputBorderColor = inputBorderColor;
var inputBgColor = (0, _utils.th)('white');
exports.inputBgColor = inputBgColor;
var inputDisabledBgColor = (0, _utils.th)('gray100');
exports.inputDisabledBgColor = inputDisabledBgColor;
var inputDisabledText = (0, _utils.th)('gray600');
exports.inputDisabledText = inputDisabledText;
var inputPlaceholderText = (0, _utils.th)('gray600');
exports.inputPlaceholderText = inputPlaceholderText;
var inputTextColor = (0, _utils.th)('gray900'); // Controls

exports.inputTextColor = inputTextColor;
var controlFocusBorderColor = (0, _utils.th)('primary', function (color) {
  return (0, _polished.lighten)(0.25, color);
});
exports.controlFocusBorderColor = controlFocusBorderColor;

var controlFocusBoxShadow = function controlFocusBoxShadow(props) {
  return function (color) {
    return (0, _styledComponents.css)(_templateObject(), (0, _utils.th)(color, function (c) {
      return (0, _polished.transparentize)(0.75, c);
    }));
  };
}; // Alerts


exports.controlFocusBoxShadow = controlFocusBoxShadow;
var alertPaddingY = '.75rem';
exports.alertPaddingY = alertPaddingY;
var alertPaddingX = '1.25rem';
exports.alertPaddingX = alertPaddingX;
var alertMarginBottom = '1rem';
exports.alertMarginBottom = alertMarginBottom;
var alertBgLevel = -10;
exports.alertBgLevel = alertBgLevel;
var alertBorderLevel = -9;
exports.alertBorderLevel = alertBorderLevel;
var alertColorLevel = 6; // Z-indexes

exports.alertColorLevel = alertColorLevel;
var zIndexControl = 1;
exports.zIndexControl = zIndexControl;
var zIndexInnerSwitch = 10;
exports.zIndexInnerSwitch = zIndexInnerSwitch;
var zIndexModal = 1050;
exports.zIndexModal = zIndexModal;
var zIndexModalBackdrop = 1071; // Transitions

exports.zIndexModalBackdrop = zIndexModalBackdrop;
var safeTransitionProperties = ['color', 'border-style', 'visibility', 'background', 'background-color', 'text-decoration', 'box-shadow', 'transform'];
var transitionBase = safeTransitionProperties.map(function (prop) {
  return prop + " .2s ease-in-out";
}).join(','); // Breakpoints

exports.transitionBase = transitionBase;
var breakpoints = _defaultBreakpoints.default; // Color levels

exports.breakpoints = breakpoints;
var yiqContrastedThreshold = 150;
exports.yiqContrastedThreshold = yiqContrastedThreshold;
var colorInterval = 0.08; // Headings

exports.colorInterval = colorInterval;
var headingsMarginBottom = '.5rem';
exports.headingsMarginBottom = headingsMarginBottom;
var headingsFontFamily = null;
exports.headingsFontFamily = headingsFontFamily;
var headingsFontWeight = 500;
exports.headingsFontWeight = headingsFontWeight;
var headingsLineHeight = 1.2;
exports.headingsLineHeight = headingsLineHeight;
var headingsColor = 'inherit';
exports.headingsColor = headingsColor;
var h1FontSize = '2.5rem';
exports.h1FontSize = h1FontSize;
var h2FontSize = '2rem';
exports.h2FontSize = h2FontSize;
var h3FontSize = '1.75rem';
exports.h3FontSize = h3FontSize;
var h4FontSize = '1.5rem';
exports.h4FontSize = h4FontSize;
var h5FontSize = '1.25rem';
exports.h5FontSize = h5FontSize;
var h6FontSize = '1rem';
exports.h6FontSize = h6FontSize;
var display1Size = '6rem';
exports.display1Size = display1Size;
var display2Size = '5.5rem';
exports.display2Size = display2Size;
var display3Size = '4.4rem';
exports.display3Size = display3Size;
var display4Size = '3.5rem';
exports.display4Size = display4Size;
var display1Weight = 300;
exports.display1Weight = display1Weight;
var display2Weight = 300;
exports.display2Weight = display2Weight;
var display3Weight = 300;
exports.display3Weight = display3Weight;
var display4Weight = 300;
exports.display4Weight = display4Weight;
var displayLineHeight = (0, _utils.th)('headingsLineHeight'); // Modals

exports.displayLineHeight = displayLineHeight;
var modalBackdropBg = 'rgba(0, 0, 0, 0.5)';
exports.modalBackdropBg = modalBackdropBg;
var modalInnerPadding = '1rem';
exports.modalInnerPadding = modalInnerPadding;
var modalTransitionDuration = 300; // ms

exports.modalTransitionDuration = modalTransitionDuration;
var modalDialogMargin = '0.5rem';
exports.modalDialogMargin = modalDialogMargin;
var modalDialogMarginYSmUp = '1.75rem';
exports.modalDialogMarginYSmUp = modalDialogMarginYSmUp;
var modalContentBg = (0, _utils.th)('white');
exports.modalContentBg = modalContentBg;
var modalContentBorderWidth = (0, _utils.th)('borderWidth');
exports.modalContentBorderWidth = modalContentBorderWidth;
var modalContentBorderColor = (0, _utils.th)('black', function (color) {
  return (0, _polished.transparentize)(0.8, color);
});
exports.modalContentBorderColor = modalContentBorderColor;
var modalContentBorderRadius = (0, _utils.th)('borderRadiusLg');
exports.modalContentBorderRadius = modalContentBorderRadius;
var modalContentBoxShadowXs = (0, _styledComponents.css)(_templateObject2(), (0, _utils.th)('black', function (color) {
  return (0, _polished.transparentize)(0.8, color);
}));
exports.modalContentBoxShadowXs = modalContentBoxShadowXs;
var modalContentBoxShadowSmUp = (0, _styledComponents.css)(_templateObject3(), (0, _utils.th)('black', function (color) {
  return (0, _polished.transparentize)(0.8, color);
}));
exports.modalContentBoxShadowSmUp = modalContentBoxShadowSmUp;
var modalHeaderBorderColor = (0, _utils.th)('gray200');
exports.modalHeaderBorderColor = modalHeaderBorderColor;
var modalFooterBorderColor = (0, _utils.th)('modalHeaderBorderColor');
exports.modalFooterBorderColor = modalFooterBorderColor;
var modalHeaderBorderWidth = (0, _utils.th)('modalContentBorderWidth');
exports.modalHeaderBorderWidth = modalHeaderBorderWidth;
var modalFooterBorderWidth = (0, _utils.th)('modalHeaderBorderWidth'); // Mixins

exports.modalFooterBorderWidth = modalFooterBorderWidth;

var base = function base(props) {
  return function () {
    return (0, _styledComponents.css)(_templateObject4(), (0, _utils.th)('fontFamily'), (0, _utils.th)('fontSizeBase'), (0, _utils.th)('lineHeightBase'));
  };
};

exports.base = base;

var controlFocus = function controlFocus(props) {
  return function (baseColor) {
    if (baseColor === void 0) {
      baseColor = 'primary';
    }

    return (0, _styledComponents.css)(_templateObject5(), (0, _utils.th)(baseColor, function (color) {
      return (0, _polished.transparentize)(0.1, color);
    }));
  };
};

exports.controlFocus = controlFocus;

var btnVariant = function btnVariant(props) {
  return function (baseColor) {
    return (0, _styledComponents.css)(_templateObject6(), (0, _utils.mixin)('colorYik', (0, _utils.th)(baseColor)), (0, _utils.th)(baseColor), (0, _utils.mixin)('controlFocus', baseColor), (0, _utils.th)(baseColor, function (color) {
      return (0, _polished.darken)(0.05, color);
    }));
  };
};

exports.btnVariant = btnVariant;

var alertVariant = function alertVariant(props) {
  return function (baseColor) {
    return (0, _styledComponents.css)(_templateObject7(), (0, _utils.mixin)('colorLevel', (0, _utils.th)(baseColor), (0, _utils.th)('alertColorLevel')), (0, _utils.mixin)('colorLevel', (0, _utils.th)(baseColor), (0, _utils.th)('alertBgLevel')), (0, _utils.mixin)('colorLevel', (0, _utils.th)(baseColor), (0, _utils.th)('alertBorderLevel')), function (props) {
      return (0, _polished.darken)(0.05, (0, _utils.mixin)('colorLevel', (0, _utils.th)(baseColor), (0, _utils.th)('alertColorLevel'))(props));
    });
  };
};

exports.alertVariant = alertVariant;

var colorLevel = function colorLevel(props) {
  return function (color, level) {
    color = typeof color === 'function' ? color(props) : color;
    level = typeof level === 'function' ? level(props) : level;
    var baseColor = level > 0 ? (0, _utils.th)('black')(props) : (0, _utils.th)('white')(props);
    var absLevel = Math.abs(level);
    return (0, _polished.mix)(absLevel * colorInterval, baseColor, color);
  };
};

exports.colorLevel = colorLevel;

var colorYik = function colorYik(props) {
  return function (color) {
    color = typeof color === 'function' ? color(props) : color;

    var _parseToRgb = (0, _polished.parseToRgb)(color),
        r = _parseToRgb.red,
        g = _parseToRgb.green,
        b = _parseToRgb.blue;

    var yik = (r * 299 + g * 587 + b * 114) / 1000;
    return yik >= (0, _utils.th)('yiqContrastedThreshold')(props) ? (0, _utils.th)('yikTextDark')(props) : (0, _utils.th)('yikTextLight')(props);
  };
};

exports.colorYik = colorYik;
