// smooth-ui/Alert.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n        &.sui-alert-", " {\n          ", ";\n        }\n      "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    position: relative;\n    padding: ", " ", ";\n    margin-bottom: ", ";\n    border: 1px solid transparent;\n    border-radius: ", ";\n\n    ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var variants = ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'light', 'dark'];
var Alert = (0, _createComponent.default)(function (_ref) {
  var th = _ref.th,
      mixin = _ref.mixin,
      css = _ref.css,
      classNames = _ref.classNames;
  return {
    name: 'alert',
    render: function render(_ref2) {
      var _classNames;

      var Component = _ref2.Component,
          className = _ref2.className,
          variant = _ref2.variant,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "variant"]);

      return _react.default.createElement(Component, _extends({}, props, {
        className: classNames(className, (_classNames = {}, _classNames["sui-alert-" + variant] = variant, _classNames))
      }));
    },
    style: css(_templateObject(), th('alertPaddingY'), th('alertPaddingX'), th('alertMarginBottom'), th('borderRadius'), variants.map(function (variant) {
      return css(_templateObject2(), variant, mixin('alertVariant', variant));
    })),
    propTypes: {
      children: _propTypes.default.node,
      variant: _propTypes.default.oneOf(variants).isRequired
    }
  };
});
var _default = Alert;
exports.default = _default;
