// smooth-ui/Box.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = require("styled-components");

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n    display: flex;\n\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n        ", ": ", ";\n      "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var addProp = function addProp(propName, attribute, transform) {
  if (transform === void 0) {
    transform = function transform(x) {
      return x;
    };
  }

  return function (props) {
    return typeof props[propName] !== 'undefined' ? (0, _styledComponents.css)(_templateObject(), attribute || propName, transform(props[propName])) : null;
  };
};

var Box = (0, _createComponent.default)(function () {
  return {
    name: 'box',
    render: function render(_ref) {
      var Component = _ref.Component,
          flex = _ref.flex,
          direction = _ref.direction,
          wrap = _ref.wrap,
          alignItems = _ref.alignItems,
          alignContent = _ref.alignContent,
          alignSelf = _ref.alignSelf,
          justifyContent = _ref.justifyContent,
          padding = _ref.padding,
          margin = _ref.margin,
          props = _objectWithoutPropertiesLoose(_ref, ["Component", "flex", "direction", "wrap", "alignItems", "alignContent", "alignSelf", "justifyContent", "padding", "margin"]);

      return _react.default.createElement(Component, props);
    },
    style: (0, _styledComponents.css)(_templateObject2(), addProp('flex', 'flex', function (flex) {
      return flex === true ? 1 : flex;
    }), addProp('direction', 'flex-direction'), addProp('wrap', 'flex-wrap'), addProp('alignItems', 'align-items'), addProp('alignContent', 'align-content'), addProp('alignSelf', 'align-self'), addProp('justifyContent', 'justify-content'), addProp('padding'), addProp('margin')),
    propTypes: {
      alignContent: _propTypes.default.string,
      alignItems: _propTypes.default.string,
      alignSelf: _propTypes.default.string,
      children: _propTypes.default.node,
      direction: _propTypes.default.string,
      flex: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number, _propTypes.default.bool]),
      justifyContent: _propTypes.default.string,
      margin: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
      padding: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
      wrap: _propTypes.default.string
    }
  };
});
var _default = Box;
exports.default = _default;
