// smooth-ui/Button.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n        &.sui-button-", " {\n          ", ";\n        }\n      "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    padding: ", " ", ";\n    z-index: ", ";\n    border-radius: ", ";\n    font-size: ", ";\n    line-height: ", ";\n    border-width: ", ";\n    cursor: pointer;\n    transition: ", ";\n\n    &.sui-button-sm {\n      padding: ", " ", ";\n      font-size: ", ";\n      border-radius: ", ";\n    }\n\n    &.sui-button-lg {\n      padding: ", " ", ";\n      font-size: ", ";\n      border-radius: ", ";\n    }\n\n    &:disabled {\n      opacity: ", ";\n    }\n\n    ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var variants = ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'light', 'dark'];
var Button = (0, _createComponent.default)(function (_ref) {
  var classNames = _ref.classNames,
      css = _ref.css,
      th = _ref.th,
      mixin = _ref.mixin;
  return {
    name: 'button',
    defaultComponent: 'button',
    render: function render(_ref2) {
      var _classNames;

      var className = _ref2.className,
          Component = _ref2.Component,
          size = _ref2.size,
          variant = _ref2.variant,
          props = _objectWithoutPropertiesLoose(_ref2, ["className", "Component", "size", "variant"]);

      return _react.default.createElement(Component, _extends({}, props, {
        className: classNames(className, (_classNames = {}, _classNames["sui-button-" + size] = size, _classNames["sui-button-" + variant] = variant, _classNames))
      }));
    },
    style: css(_templateObject(), th('btnPaddingY'), th('btnPaddingX'), th('zIndexControl'), th('borderRadius'), th('fontSizeBase'), th('lineHeightBase'), th('btnBorderWidth'), th('transitionBase'), th('btnPaddingYSm'), th('btnPaddingXSm'), th('fontSizeSm'), th('borderRadiusSm'), th('btnPaddingYLg'), th('btnPaddingXLg'), th('fontSizeLg'), th('borderRadiusLg'), th('btnDisabledOpacity'), variants.map(function (variant) {
      return css(_templateObject2(), variant, mixin('btnVariant', variant));
    })),
    propTypes: {
      children: _propTypes.default.node,
      disabled: _propTypes.default.bool,
      size: _propTypes.default.oneOf(['sm', 'lg']),
      variant: _propTypes.default.oneOf(variants)
    },
    defaultProps: {
      variant: 'primary'
    }
  };
});
var _default = Button;
exports.default = _default;
