// smooth-ui/Checkbox.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _InnerSwitch = _interopRequireDefault(require("./internal/InnerSwitch"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    position: relative;\n    width: 1.5rem;\n    height: 1.5rem;\n    z-index: ", ";\n\n    .sui-checkbox-content {\n      display: flex;\n      align-items: center;\n      justify-content: center;\n      width: 1rem;\n      height: 1rem;\n      background-color: ", ";\n      border-radius: ", ";\n      border-style: solid;\n      border-width: ", ";\n      border-color: ", ";\n      transition: ", ";\n\n      &.checked {\n        background-color: ", ";\n        border-color: transparent;\n\n        svg {\n          transform: scale(1);\n        }\n      }\n\n      &.focused {\n        ", ";\n      }\n\n      &.disabled {\n        background-color: ", ";\n      }\n    }\n\n    svg {\n      width: 0.75rem;\n      pointer-events: none;\n      transform: scale(0);\n      transition: ", ";\n    }\n\n    &.sui-checkbox-sm {\n      .sui-checkbox-content {\n        border-radius: ", ";\n        width: 0.875rem;\n        height: 0.875rem;\n      }\n\n      svg {\n        width: 0.65rem;\n      }\n    }\n\n    &.sui-checkbox-lg {\n      .sui-checkbox-content {\n        border-radius: ", ";\n        width: 1.25rem;\n        height: 1.25rem;\n      }\n\n      svg {\n        width: 0.9375rem;\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Checkbox = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      mixin = _ref.mixin,
      th = _ref.th,
      classNames = _ref.classNames;
  return {
    name: 'checkbox',
    render: function render(_ref2) {
      var _classNames;

      var Component = _ref2.Component,
          className = _ref2.className,
          size = _ref2.size,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "size"]);

      return _react.default.createElement(Component, {
        className: classNames(className, (_classNames = {
          'sui-checkbox-disabled': props.disabled
        }, _classNames["sui-checkbox-" + size] = size, _classNames))
      }, _react.default.createElement(_InnerSwitch.default, _extends({
        inputType: "checkbox"
      }, props), function (_ref3) {
        var checked = _ref3.checked,
            focused = _ref3.focused,
            disabled = _ref3.disabled;
        return _react.default.createElement("div", {
          className: classNames('sui-checkbox-content', {
            checked: checked,
            focused: focused,
            disabled: disabled
          })
        }, _react.default.createElement("svg", {
          viewBox: "0 0 10 8"
        }, _react.default.createElement("path", {
          d: "M3.7 7.3L.3 4l1-.8 2.4 2.3 5-4.8 1 1",
          fill: "#fff",
          fillRule: "evenodd"
        })));
      }));
    },
    style: css(_templateObject(), th('zIndexControl'), th('inputBgColor'), th('borderRadius'), th('inputBorderWidth'), th('inputBorderColor'), th('transitionBase'), th('primary'), mixin('controlFocus'), th('inputDisabledBgColor'), th('transitionBase'), th('borderRadiusSm'), th('borderRadiusLg')),
    propTypes: {
      checked: _propTypes.default.bool,
      disabled: _propTypes.default.bool,
      onChange: _propTypes.default.func,
      size: _propTypes.default.oneOf(['sm', 'lg']),
      value: _propTypes.default.string
    }
  };
});
var _default = Checkbox;
exports.default = _default;
