// smooth-ui/Col.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = require("styled-components");

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n    flex-basis: 0;\n    flex-grow: 1;\n    max-width: 100%;\n    ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n        &.sui-col-", "-", " {\n          position: relative;\n          width: 100%;\n          min-height: 1px;\n\n          @media (min-width: ", "px) {\n              flex: 0 0 ", ";\n              max-width: ", ";\n              ", "\n            }\n        }\n      "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    &.sui-col-", " {\n      position: relative;\n      width: 100%;\n      min-height: 1px;\n\n      @media (min-width: ", "px) {\n        flex-basis: 0;\n        flex-grow: 1;\n        max-width: 100%;\n      }\n    }\n\n    ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var GRID_SIZES = [true, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 'auto'];

function generateBreakPoint(breakpoint) {
  return (0, _styledComponents.css)(_templateObject(), breakpoint, function (props) {
    return props.theme.breakpoints[breakpoint];
  }, GRID_SIZES.map(function (size) {
    if (typeof size === 'boolean') return null; // Only keep 6 significant numbers.

    var width = size !== 'auto' ? Math.round(size / 12 * Math.pow(10, 6)) / Math.pow(10, 4) + "%" : 'auto';
    return (0, _styledComponents.css)(_templateObject2(), breakpoint, size, function (props) {
      return props.theme.breakpoints[breakpoint];
    }, width, width === 'auto' ? 'none' : width, width === 'auto' ? 'width: auto;' : '');
  }));
}

var Col = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      classNames = _ref.classNames;
  return {
    name: 'col',
    render: function render(_ref2) {
      var _classNames;

      var Component = _ref2.Component,
          className = _ref2.className,
          xs = _ref2.xs,
          sm = _ref2.sm,
          md = _ref2.md,
          lg = _ref2.lg,
          xl = _ref2.xl,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "xs", "sm", "md", "lg", "xl"]);

      return _react.default.createElement(Component, _extends({
        className: classNames(className, (_classNames = {
          'sui-col-xs': xs === true
        }, _classNames["sui-col-xs-" + String(xs)] = xs && xs !== true, _classNames['sui-col-sm'] = sm === true, _classNames["sui-col-sm-" + String(sm)] = sm && sm !== true, _classNames['sui-col-md'] = md === true, _classNames["sui-col-md-" + String(md)] = md && md !== true, _classNames['sui-col-lg'] = lg === true, _classNames["sui-col-lg-" + String(lg)] = lg && lg !== true, _classNames['sui-col-xl'] = xl === true, _classNames["sui-col-xl-" + String(xl)] = xl && xl !== true, _classNames))
      }, props));
    },
    style: css(_templateObject3(), function (props) {
      return Object.keys(props.theme.breakpoints).map(generateBreakPoint);
    }),
    propTypes: {
      children: _propTypes.default.node,
      sm: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string, _propTypes.default.number]),
      xs: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string, _propTypes.default.number]),
      md: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string, _propTypes.default.number]),
      lg: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string, _propTypes.default.number]),
      xl: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.string, _propTypes.default.number])
    }
  };
});
var _default = Col;
exports.default = _default;
