// smooth-ui/ControlFeedback.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    width: 100%;\n    margin-top: 0.25rem;\n    font-size: 80%;\n\n    &.sui-is-valid {\n      color: ", ";\n    }\n\n    &.sui-is-invalid {\n      color: ", ";\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ControlFeedback = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      classNames = _ref.classNames,
      th = _ref.th;
  return {
    name: 'control-feedback',
    render: function render(_ref2) {
      var Component = _ref2.Component,
          className = _ref2.className,
          valid = _ref2.valid,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "valid"]);

      return _react.default.createElement(Component, _extends({
        className: classNames(className, {
          'sui-is-valid': valid === true,
          'sui-is-invalid': valid === false
        }, className)
      }, props));
    },
    style: css(_templateObject(), th('success'), th('danger')),
    propTypes: {
      children: _propTypes.default.node,
      valid: _propTypes.default.bool.isRequired
    }
  };
});
var _default = ControlFeedback;
exports.default = _default;
