// smooth-ui/Input.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    border-width: ", ";\n    border-color: ", ";\n    border-style: solid;\n    border-radius: ", ";\n    padding: ", " ", ";\n    font-size: ", ";\n    line-height: ", ";\n    color: ", ";\n    transition: ", ";\n    background-color: ", ";\n\n    &[type='number'] {\n      padding-right: 6px;\n    }\n\n    &::placeholder {\n      color: ", ";\n    }\n\n    &:focus {\n      ", ";\n    }\n\n    &[disabled] {\n      background-color: ", ";\n      color: ", ";\n    }\n\n    &.sui-input-sm {\n      padding: ", " ", ";\n      font-size: ", ";\n      border-radius: ", ";\n    }\n\n    &.sui-input-lg {\n      padding: ", " ", ";\n      font-size: ", ";\n      border-radius: ", ";\n    }\n\n    &.sui-control {\n      display: block;\n      width: 100%;\n\n      &:focus {\n        border-color: ", ";\n        box-shadow: ", ";\n      }\n\n      &.sui-is-valid {\n        border-color: ", ";\n\n        &:focus {\n          border-color: ", ";\n          box-shadow: ", ";\n        }\n      }\n\n      &.sui-is-invalid {\n        border-color: ", ";\n\n        &:focus {\n          border-color: ", ";\n          box-shadow: ", ";\n        }\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Input = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      th = _ref.th,
      mixin = _ref.mixin,
      classNames = _ref.classNames;
  return {
    name: 'input',
    defaultComponent: 'input',
    render: function render(_ref2) {
      var _classNames;

      var Component = _ref2.Component,
          className = _ref2.className,
          control = _ref2.control,
          size = _ref2.size,
          valid = _ref2.valid,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "control", "size", "valid"]);

      return _react.default.createElement(Component, _extends({}, props, {
        className: classNames(className, (_classNames = {
          'sui-control': control,
          'sui-is-valid': valid === true,
          'sui-is-invalid': valid === false
        }, _classNames["sui-input-" + size] = size, _classNames))
      }));
    },
    style: css(_templateObject(), th('inputBorderWidth'), th('inputBorderColor'), th('borderRadius'), th('inputPaddingY'), th('inputPaddingX'), th('fontSizeBase'), th('inputLineHeight'), th('inputTextColor'), th('transitionBase'), th('inputBgColor'), th('inputPlaceholderText'), mixin('controlFocus'), th('inputDisabledBgColor'), th('inputDisabledText'), th('inputPaddingYSm'), th('inputPaddingXSm'), th('fontSizeSm'), th('borderRadiusSm'), th('inputPaddingYLg'), th('inputPaddingXLg'), th('fontSizeLg'), th('borderRadiusLg'), th('controlFocusBorderColor'), mixin('controlFocusBoxShadow', 'primary'), th('success'), th('success'), mixin('controlFocusBoxShadow', 'success'), th('danger'), th('danger'), mixin('controlFocusBoxShadow', 'danger')),
    propTypes: {
      control: _propTypes.default.bool,
      size: _propTypes.default.oneOf(['sm', 'lg']),
      valid: _propTypes.default.bool
    }
  };
});
var _default = Input;
exports.default = _default;
