// smooth-ui/Modal.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactDom = require("react-dom");

var _classnames = _interopRequireDefault(require("classnames"));

var _Transition = _interopRequireDefault(require("./Transition"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

var _utils = require("./utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    position: fixed;\n    top: 0;\n    right: 0;\n    bottom: 0;\n    left: 0;\n    z-index: ", ";\n    visibility: hidden;\n    overflow: hidden;\n    outline: 0;\n    transition: opacity ", "ms ease-in-out;\n\n    &.sui-modal-opened {\n      visibility: visible;\n      overflow-x: hidden;\n      overflow-y: auto;\n    }\n\n    &.sui-modal-transition-entering {\n      opacity: 1;\n    }\n\n    &.sui-modal-transition-entered {\n      opacity: 1;\n    }\n\n    &.sui-modal-transition-exited {\n      opacity: 0;\n    }\n\n    &.sui-modal-transition-exiting {\n      opacity: 0;\n    }\n\n    .sui-modal-backdrop {\n      position: fixed;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n      background-color: ", ";\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var ModalComponent =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(ModalComponent, _React$Component);

  function ModalComponent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;

    _this.handleKeyup = function (_ref) {
      var keyCode = _ref.keyCode;

      if (keyCode === 27
      /* Escape */
      ) {
          _this.props.onClose();
        }
    };

    if (!_this.container && typeof document !== 'undefined') {
      _this.container = document.createElement('div');
      document.body.appendChild(_this.container);
    }

    return _this;
  }

  var _proto = ModalComponent.prototype;

  _proto.componentWillUnmount = function componentWillUnmount() {
    document.body.removeChild(this.container);
    document.removeEventListener('keyup', this.handleKeyup);
  };

  _proto.setupKeyHandling = function setupKeyHandling() {
    if (this.props.opened) {
      document.body.style.overflow = 'hidden';
      document.addEventListener('keyup', this.handleKeyup);
    } else {
      document.body.style.overflow = null;
      document.removeEventListener('keyup', this.handleKeyup);
    }
  };

  _proto.componentDidMount = function componentDidMount() {
    this.setupKeyHandling();
  };

  _proto.componentDidUpdate = function componentDidUpdate() {
    this.setupKeyHandling();
  };

  _proto.render = function render() {
    var _this$props = this.props,
        className = _this$props.className,
        theme = _this$props.theme,
        opened = _this$props.opened,
        onClose = _this$props.onClose,
        children = _this$props.children,
        props = _objectWithoutPropertiesLoose(_this$props, ["className", "theme", "opened", "onClose", "children"]);

    if (!this.container) return null;
    return (0, _reactDom.createPortal)(_react.default.createElement(_Transition.default, {
      timeout: theme.modalTransitionDuration,
      in: this.props.opened
    }, function (transitionState) {
      var _classNames;

      return _react.default.createElement("div", _extends({
        role: "dialog",
        tabIndex: "-1",
        className: (0, _classnames.default)('sui-modal', (_classNames = {
          'sui-modal-opened': opened || transitionState === 'exiting'
        }, _classNames["sui-modal-transition-" + transitionState] = transitionState, _classNames), className)
      }, props), _react.default.createElement("div", {
        className: "sui-modal-backdrop",
        onClick: onClose
      }), children);
    }), this.container);
  };

  return ModalComponent;
}(_react.default.Component);

var Modal = (0, _createComponent.default)(function (_ref2) {
  var css = _ref2.css;
  return {
    name: 'modal',
    InnerComponent: ModalComponent,
    style: css(_templateObject(), (0, _utils.th)('zIndexModal'), (0, _utils.th)('modalTransitionDuration'), (0, _utils.th)('modalBackdropBg')),
    propTypes: {
      children: _propTypes.default.node,
      opened: _propTypes.default.bool,
      onClose: _propTypes.default.func
    }
  };
});
var _default = Modal;
exports.default = _default;
