// smooth-ui/ModalDialog.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n        max-width: 500px;\n        margin: ", " auto;\n        min-height: calc(\n          100% -\n            ", "\n        );\n      "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    position: relative;\n    width: auto;\n    margin: ", ";\n    pointer-events: none;\n\n    /* center modal */\n    display: flex;\n    align-items: center;\n    min-height: calc(\n      100% - ", "\n    );\n\n    ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ModalDialog = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      th = _ref.th,
      calc = _ref.calc,
      up = _ref.up;
  return {
    name: 'modal-dialog',
    render: function render(_ref2) {
      var Component = _ref2.Component,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component"]);

      return _react.default.createElement(Component, _extends({
        role: "document"
      }, props));
    },
    style: css(_templateObject(), th('modalDialogMargin'), th('modalDialogMargin', function (value) {
      return calc(value, function (x) {
        return x * 2;
      });
    }), up('sm', css(_templateObject2(), th('modalDialogMarginYSmUp'), th('modalDialogMarginYSmUp', function (value) {
      return calc(value, function (x) {
        return x * 2;
      });
    })))),
    propTypes: {
      children: _propTypes.default.node
    }
  };
});
var _default = ModalDialog;
exports.default = _default;
