// smooth-ui/RadioGroup.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var RadioGroup =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(RadioGroup, _React$Component);

  function RadioGroup() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.controls = [];
    return _this;
  }

  var _proto = RadioGroup.prototype;

  _proto.getChildContext = function getChildContext() {
    return {
      suiGroup: this
    };
  };

  _proto.register = function register(control) {
    this.controls.push(control);
  };

  _proto.unregister = function unregister(control) {
    var index = this.controls.indexOf(control);
    if (index !== -1) this.controls.splice(index, 1);
  };

  _proto.notify = function notify(event) {
    this.controls.forEach(function (control) {
      if (control.props.checked === undefined && control.props.name === event.target.name && control.props.value !== event.target.value) {
        control.updateState({
          checked: false
        });
      }
    });
  };

  _proto.render = function render() {
    return this.props.children;
  };

  return RadioGroup;
}(_react.default.Component);

RadioGroup.childContextTypes = {
  suiGroup: _propTypes.default.object
};
RadioGroup.propTypes = {
  children: _propTypes.default.node
};
var _default = RadioGroup;
exports.default = _default;
