// smooth-ui/Radio.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _InnerSwitch = _interopRequireDefault(require("./internal/InnerSwitch"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    width: 1.5rem;\n    height: 1.5rem;\n    position: relative;\n\n    .sui-radio-content {\n      display: flex;\n      flex-shrink: 0;\n      align-items: center;\n      justify-content: center;\n      width: 1rem;\n      height: 1rem;\n      border-radius: 50%;\n      background-color: ", ";\n      border-width: ", ";\n      border-style: solid;\n      border-color: ", ";\n      transition: ", ";\n\n      &.checked {\n        border-color: ", ";\n\n        .sui-radio-circle {\n          transform: scale(1);\n        }\n      }\n\n      &.focused {\n        ", ";\n      }\n\n      &.disabled {\n        background-color: ", ";\n      }\n    }\n\n    .sui-radio-circle {\n      width: 0.6rem;\n      height: 0.6rem;\n      transition: ", ";\n      border-radius: 50%;\n      background-color: ", ";\n      transform: scale(0);\n    }\n\n    &.sui-radio-sm {\n      .sui-radio-content {\n        width: 0.875rem;\n        height: 0.875rem;\n      }\n\n      .sui-radio-circle {\n        width: 0.525rem;\n        height: 0.525rem;\n      }\n    }\n\n    &.sui-radio-lg {\n      .sui-radio-content {\n        width: 1.25rem;\n        height: 1.25rem;\n      }\n\n      .sui-radio-circle {\n        width: 0.75rem;\n        height: 0.75rem;\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ModalHeader = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      th = _ref.th,
      mixin = _ref.mixin,
      classNames = _ref.classNames;
  return {
    name: 'radio',
    render: function render(_ref2) {
      var _classNames;

      var Component = _ref2.Component,
          className = _ref2.className,
          size = _ref2.size,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "size"]);

      return _react.default.createElement(Component, {
        className: classNames(className, (_classNames = {
          'sui-radio-disabled': props.disabled
        }, _classNames["sui-radio-" + size] = size, _classNames))
      }, _react.default.createElement(_InnerSwitch.default, _extends({
        inputType: "radio"
      }, props), function (_ref3) {
        var focused = _ref3.focused,
            checked = _ref3.checked,
            disabled = _ref3.disabled;
        return _react.default.createElement("div", {
          className: classNames('sui-radio-content', {
            focused: focused,
            checked: checked,
            disabled: disabled
          })
        }, _react.default.createElement("div", {
          className: "sui-radio-circle"
        }));
      }));
    },
    style: css(_templateObject(), th('inputBgColor'), th('inputBorderWidth'), th('inputBorderColor'), th('transitionBase'), th('primary'), mixin('controlFocus'), th('inputDisabledBgColor'), th('transitionBase'), th('primary')),
    propTypes: {
      size: _propTypes.default.oneOf(['sm', 'lg'])
    }
  };
});
var _default = ModalHeader;
exports.default = _default;
