// smooth-ui/Row.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    flex-grow: 1;\n    flex-wrap: wrap;\n    display: flex;\n    margin-right: -", ";\n    margin-left: -", ";\n\n    > .sui-col {\n      padding-left: ", ";\n      padding-right: ", ";\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Row = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      resolveUnit = _ref.resolveUnit;
  return {
    name: 'row',
    render: function render(_ref2) {
      var Component = _ref2.Component,
          gutter = _ref2.gutter,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "gutter"]);

      return _react.default.createElement(Component, props);
    },
    style: css(_templateObject(), function (props) {
      return resolveUnit(props.gutter);
    }, function (props) {
      return resolveUnit(props.gutter);
    }, function (props) {
      return resolveUnit(props.gutter);
    }, function (props) {
      return resolveUnit(props.gutter);
    }),
    propTypes: {
      children: _propTypes.default.node
    },
    defaultProps: {
      gutter: 15
    }
  };
});
var _default = Row;
exports.default = _default;
