// smooth-ui/Select.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    position: relative;\n\n    select {\n      /* Disable appearance */\n      -webkit-appearance: none;\n      -moz-appearance: none;\n      -webkit-border-radius: 0;\n\n      display: inline-block;\n      border-width: ", ";\n      border-color: ", ";\n      border-style: solid;\n      border-radius: ", ";\n      padding: ", " ", ";\n      font-size: ", ";\n      line-height: ", ";\n      transition: ", ";\n      color: ", ";\n      ", ";\n\n      &:focus {\n        ", ";\n      }\n\n      &[disabled] {\n        background-color: ", ";\n        color: ", ";\n      }\n    }\n\n    .sui-select-arrow {\n      position: absolute;\n      top: 50%;\n      right: 0.5rem;\n      width: 0.6rem;\n      pointer-events: none;\n    }\n\n    &.sui-select-sm {\n      select {\n        padding: ", " ", ";\n        font-size: ", ";\n        border-radius: ", ";\n        ", ";\n      }\n\n      .sui-select-arrow {\n        right: 0.35rem;\n        width: 0.525rem;\n      }\n    }\n\n    &.sui-select-lg {\n      select {\n        padding: ", " ", ";\n        font-size: ", ";\n        border-radius: ", ";\n        ", ";\n      }\n\n      .sui-select-arrow {\n        right: 0.625rem;\n        width: 0.75rem;\n      }\n    }\n\n    &.sui-control {\n      display: block;\n      width: 100%;\n    }\n\n    &.sui-control {\n      display: block;\n      width: 100%;\n\n      select {\n        display: block;\n        width: 100%;\n\n        &:focus {\n          border-color: ", ";\n          box-shadow: ", ";\n        }\n      }\n\n      &.sui-is-valid select {\n        border-color: ", ";\n\n        &:focus {\n          border-color: ", ";\n          box-shadow: ", ";\n        }\n      }\n\n      &.sui-is-invalid select {\n        border-color: ", ";\n\n        &:focus {\n          border-color: ", ";\n          box-shadow: ", ";\n        }\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var renderOption = function renderOption(option) {
  var _ref = typeof option === 'string' ? {
    label: option,
    value: option
  } : option,
      label = _ref.label,
      value = _ref.value;

  return _react.default.createElement("option", {
    key: value,
    value: value
  }, label);
};

var Select = (0, _createComponent.default)(function (_ref2) {
  var css = _ref2.css,
      classNames = _ref2.classNames,
      th = _ref2.th,
      mixin = _ref2.mixin;
  return {
    name: 'select',
    render: function render(_ref3) {
      var _classNames;

      var Component = _ref3.Component,
          arrow = _ref3.arrow,
          control = _ref3.control,
          className = _ref3.className,
          options = _ref3.options,
          size = _ref3.size,
          valid = _ref3.valid,
          props = _objectWithoutPropertiesLoose(_ref3, ["Component", "arrow", "control", "className", "options", "size", "valid"]);

      return _react.default.createElement(Component, {
        className: classNames(className, (_classNames = {
          'sui-control': control,
          'sui-is-valid': valid === true,
          'sui-is-invalid': valid === false
        }, _classNames["sui-select-" + size] = size, _classNames))
      }, arrow && !props.multiple ? _react.default.createElement("svg", {
        className: "sui-select-arrow",
        viewBox: "0 0 10 5"
      }, _react.default.createElement("g", {
        fill: "none",
        fillRule: "evenodd"
      }, _react.default.createElement("path", {
        d: "M17 14H-7v-24h24"
      }), _react.default.createElement("path", {
        fill: "currentColor",
        opacity: 0.5,
        d: "M0 0l5 5 5-5"
      }))) : null, _react.default.createElement("select", props, options.map(function (node) {
        return node.options ? _react.default.createElement("optgroup", {
          key: node.label,
          label: node.label
        }, node.options.map(renderOption)) : renderOption(node);
      })));
    },
    style: css(_templateObject(), th('inputBorderWidth'), th('inputBorderColor'), th('borderRadius'), th('inputPaddingY'), th('inputPaddingX'), th('fontSizeBase'), th('lineHeightBase'), th('transitionBase'), th('inputTextColor'), function (props) {
      return props.arrow && !props.multiple && 'padding-right: 1.6rem;';
    }, mixin('controlFocus'), th('inputDisabledBgColor'), th('inputDisabledText'), th('inputPaddingYSm'), th('inputPaddingXSm'), th('fontSizeSm'), th('borderRadiusSm'), function (props) {
      return props.arrow && !props.multiple && 'padding-right: 1.225rem;';
    }, th('inputPaddingYLg'), th('inputPaddingXLg'), th('fontSizeLg'), th('borderRadiusLg'), function (props) {
      return props.arrow && !props.multiple && 'padding-right: 2rem;';
    }, th('controlFocusBorderColor'), mixin('controlFocusBoxShadow', 'primary'), th('success'), th('success'), mixin('controlFocusBoxShadow', 'success'), th('danger'), th('danger'), mixin('controlFocusBoxShadow', 'danger')),
    propTypes: {
      children: _propTypes.default.node
    }
  };
});
var _default = Select;
exports.default = _default;
