// smooth-ui/Switch.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _InnerSwitch = _interopRequireDefault(require("./internal/InnerSwitch"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    display: inline-block;\n    position: relative;\n    width: 50px;\n    height: 24px;\n\n    .sui-switch-wrapper {\n      width: 50px;\n      height: 24px;\n      border-radius: 34px;\n      background-color: ", ";\n      overflow: hidden;\n      cursor: pointer;\n      border-width: ", ";\n      border-color: ", ";\n      border-style: solid;\n      transition: ", ";\n\n      &.focused {\n        ", ";\n      }\n\n      &.checked {\n        .sui-switch-content {\n          transform: translateX(0);\n        }\n\n        background-color: ", ";\n        border-color: transparent;\n\n        .sui-switch-ball {\n          background-color: ", ";\n        }\n      }\n\n      &.disabled {\n        opacity: 0.5;\n      }\n    }\n\n    .sui-switch-ball {\n      flex-shrink: 0;\n      background-color: ", ";\n      border-radius: 50%;\n      width: 18px;\n      height: 18px;\n      transition: ", ";\n    }\n\n    .sui-switch-content {\n      display: flex;\n      align-items: center;\n      height: 22px;\n      transition: ", ";\n      transform: translateX(-25px);\n    }\n\n    .sui-switch-label {\n      flex-shrink: 0;\n      font-size: 9px;\n      font-weight: 800;\n      width: 27px;\n      text-align: center;\n      user-select: none;\n\n      &.on {\n        color: ", ";\n      }\n\n      &.off {\n        color: ", ";\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Switch = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      mixin = _ref.mixin,
      th = _ref.th,
      classNames = _ref.classNames;
  return {
    name: 'switch',
    render: function render(_ref2) {
      var Component = _ref2.Component,
          className = _ref2.className,
          labeled = _ref2.labeled,
          props = _objectWithoutPropertiesLoose(_ref2, ["Component", "className", "labeled"]);

      return _react.default.createElement(Component, {
        className: classNames(className, {
          'sui-switch-disabled': props.disabled
        })
      }, _react.default.createElement(_InnerSwitch.default, _extends({
        inputType: "checkbox"
      }, props), function (_ref3) {
        var checked = _ref3.checked,
            focused = _ref3.focused,
            disabled = _ref3.disabled;
        return _react.default.createElement("div", {
          className: classNames('sui-switch-wrapper', {
            checked: checked,
            focused: focused,
            disabled: disabled
          })
        }, _react.default.createElement("div", {
          className: "sui-switch-content"
        }, _react.default.createElement("span", {
          className: "sui-switch-label on"
        }, labeled ? 'ON' : ''), _react.default.createElement("div", {
          className: "sui-switch-ball"
        }), _react.default.createElement("span", {
          className: "sui-switch-label off"
        }, labeled ? 'OFF' : '')));
      }));
    },
    style: css(_templateObject(), th('gray300'), th('inputBorderWidth'), th('inputBorderColor'), th('transitionBase'), mixin('controlFocus'), th('primaryLight'), th('primary'), th('gray500'), th('transitionBase'), th('transitionBase'), th('primary'), th('gray900')),
    propTypes: {
      /** Add ON/OFF labels. */
      labeled: _propTypes.default.bool
    }
  };
});
var _default = Switch;
exports.default = _default;
