// smooth-ui/Toggler.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var Toggler =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(Toggler, _React$Component);

  function Toggler() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.state = {
      toggled: _this.props.defaultToggled
    };

    _this.onToggle = function (value) {
      _this.setState(function (previousState) {
        return {
          toggled: typeof value === 'boolean' ? value : !previousState.toggled
        };
      }, function () {
        if (_this.props.onToggle) {
          _this.props.onToggle(_this.state.toggled);
        }
      });
    };

    return _this;
  }

  var _proto = Toggler.prototype;

  _proto.render = function render() {
    return this.props.children({
      toggled: this.state.toggled,
      onToggle: this.onToggle
    });
  };

  return Toggler;
}(_react.default.Component);

Toggler.propTypes = {
  children: _propTypes.default.func.isRequired,
  defaultToggled: _propTypes.default.bool,
  onToggle: _propTypes.default.func
};
Toggler.defaultProps = {
  defaultToggled: false
};
var _default = Toggler;
exports.default = _default;
