// smooth-ui/Typography.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createComponent = _interopRequireDefault(require("./internal/createComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n    &.sui-typo-h1,\n    &.sui-typo-h2,\n    &.sui-typo-h3,\n    &.sui-typo-h4,\n    &.sui-typo-h5,\n    &.sui-typo-h6,\n    &.sui-typo-display-1,\n    &.sui-typo-display-2,\n    &.sui-typo-display-3,\n    &.sui-typo-display-4 {\n      margin-top: 0;\n      margin-bottom: ", ";\n      font-weight: ", ";\n      line-height: ", ";\n      color: ", ";\n      ", ";\n\n      ", ";\n    }\n\n    &.sui-typo-h1 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-h2 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-h3 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-h4 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-h5 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-h6 {\n      font-size: ", ";\n    }\n\n    &.sui-typo-display-1 {\n      font-size: ", ";\n      font-weight: ", ";\n      line-height: ", ";\n    }\n\n    &.sui-typo-display-2 {\n      font-size: ", ";\n      font-weight: ", ";\n      line-height: ", ";\n    }\n\n    &.sui-typo-display-3 {\n      font-size: ", ";\n      font-weight: ", ";\n      line-height: ", ";\n    }\n\n    &.sui-typo-display-4 {\n      font-size: ", ";\n      font-weight: ", ";\n      line-height: ", ";\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var variantTags = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
  h5: 'h5',
  h6: 'h6',
  'display-1': 'h1',
  'display-2': 'h2',
  'display-3': 'h3',
  'display-4': 'h4'
};
var Typography = (0, _createComponent.default)(function (_ref) {
  var css = _ref.css,
      th = _ref.th,
      classNames = _ref.classNames;
  return {
    name: 'typo',
    defaultComponent: null,
    render: function render(_ref2) {
      var className = _ref2.className,
          BaseComponent = _ref2.Component,
          margin = _ref2.margin,
          variant = _ref2.variant,
          props = _objectWithoutPropertiesLoose(_ref2, ["className", "Component", "margin", "variant"]);

      var Component = BaseComponent || variantTags[variant] || 'span';
      return _react.default.createElement(Component, _extends({
        className: classNames(className, "sui-typo-" + variant)
      }, props));
    },
    style: css(_templateObject(), th('headingsMarginBottom'), th('headingsFontWeight'), th('headingsLineHeight'), th('headingsColor'), th('headingsFontFamily', function (font) {
      return font ? "font-family: " + font + ";" : null;
    }), function (props) {
      return props.margin === false && 'margin-bottom: 0;';
    }, th('h1FontSize'), th('h2FontSize'), th('h3FontSize'), th('h4FontSize'), th('h5FontSize'), th('h6FontSize'), th('display1Size'), th('display1Weight'), th('displayLineHeight'), th('display2Size'), th('display2Weight'), th('displayLineHeight'), th('display3Size'), th('display3Weight'), th('displayLineHeight'), th('display4Size'), th('display4Weight'), th('displayLineHeight')),
    propTypes: {
      children: _propTypes.default.node,
      margin: _propTypes.default.bool,
      variant: _propTypes.default.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'display-1', 'display-2', 'display-3', 'display-4'])
    },
    defaultProps: {
      margin: true
    }
  };
});
var _default = Typography;
exports.default = _default;
